subsys/bluetooth/host/gatt.c

```diff
@@ -5498,17 +5498,14 @@ void bt_gatt_connected(struct bt_conn *conn)
 		}
 	}
 
-#if defined(CONFIG_BT_GATT_CLIENT)
-	add_subscriptions(conn);
-#if defined(CONFIG_BT_GATT_AUTO_UPDATE_MTU)
+#if defined(CONFIG_BT_GATT_AUTO_UPDATE_MTU) && defined(CONFIG_BT_GATT_CLIENT)
```

Comment:
`BT_GATT_CLIENT` depends on `BT_GATT_CLIENT`, so no need for the `&& defined(CONFIG_BT_GATT_CLIENT)`.